src/components/Provider.js

```diff
@@ -35,7 +35,13 @@ export default class Provider extends React.Component<Props> {
 
     const {apiKey, children, ...options} = this.props;
 
-    this._stripe = window.Stripe(apiKey, options);
+    window.Stripe.__cachedInstances = window.Stripe.__cachedInstances || {};
+    const cacheKey = `key=${apiKey} options=${JSON.stringify(options)}`;
```

Comment:
Are there any cases where using `JSON.stringify` bites us? I think that it should be fine if the only available `options` map keys to strings. But for example:

```js
> JSON.stringify({foo: undefined});
'{}'
```

So in a case like this, we'd forget the `foo`.

Reply:
Added a comment about this not being a generic memoization solution!